ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLog.java

```diff
@@ -313,11 +338,15 @@ protected void validateLogEntry(LogEntryProto entry) {
     }
     TermIndex lastTermIndex = getLastEntryTermIndex();
     if (lastTermIndex != null) {
+      long latestSnapshotIndex = getSnapshotIndex();
+      long lastIndex = lastTermIndex.getIndex() > latestSnapshotIndex ?
+          lastTermIndex.getIndex() : latestSnapshotIndex;
```

Comment:
So the case when snapshotIndex is greater than entry but there is no lastTermIndex?
This would mean that follower has conveyed the wrong snapshotIndex/ nextIndex to the leader. 
I can add a check in validateLogEntry for this scenario but the problem would be bigger if this could happen, right?

Reply:
Suppose a scenario : Snapshot is getting installed at index 50 and log entry is getting appended at index 40 or 60. Is this case possible today?

Reply:
If a snapshot installation is in progress, append log entries would fail. RaftServerImpl#checkInconsistentAppendEntries() takes care of this. So even though the above scenario is possible, it will be taken care of before it reached RaftLog#validateLogEntry().

Reply:
Added a check to be on safe side.